src/Uno.UWP/Storage/Pickers/FolderPicker.macOS.cs

```diff
@@ -1,38 +1,32 @@
-﻿#if __MACOS__
+﻿#nullable enable
+
 using System;
+using System.Threading;
 using System.Threading.Tasks;
 using AppKit;
-using Windows.Foundation;
-
-namespace Windows.Storage.Pickers
-{
-	public partial class FolderPicker
-	{
-		public FolderPicker()
-		{	
-		}
-
-		public IAsyncOperation<StorageFolder> PickSingleFolderAsync() =>
-			PickSingleFolderImplAsync().AsAsyncOperation();
-
-		private async Task<StorageFolder> PickSingleFolderImplAsync()
-		{
-			var openPanel = new NSOpenPanel();
-			openPanel.AllowedFileTypes = new[] { "none" };
-			openPanel.AllowsOtherFileTypes = false;
-			openPanel.CanChooseDirectories = true;
-			openPanel.CanChooseFiles = false;
-			var response = openPanel.RunModal();
-			if (response == 1 && openPanel.Urls.Length > 0)
-			{
-				var path = openPanel.Urls[0]?.Path;
-				if (path != null)
-				{
-					return new StorageFolder(path);
-				}
-			}
-			return null;
-		}
-	}
-}
-#endif
\ No newline at end of file
+using Windows.Foundation;
+
+namespace Windows.Storage.Pickers
+{
+	public partial class FolderPicker
+	{
+		private async Task<StorageFolder?> PickSingleFolderTaskAsync(CancellationToken token)
+		{
+			var openPanel = new NSOpenPanel();
+			openPanel.AllowedFileTypes = new[] { "none" };
+			openPanel.AllowsOtherFileTypes = false;
+			openPanel.CanChooseDirectories = true;
+			openPanel.CanChooseFiles = false;
+			var response = openPanel.RunModal();
+			if (response == 1 && openPanel.Urls.Length > 0)

```

Comment:
Didn't understand the `1` but I see that Xamarin hasn't yet mapped the `struct` returned from this method in the Apple implementation `NSApplication.ModalResponse`

Reply:
I have moved it into a constant at least :-) .